### PR TITLE
soundfile: Fix build on aarch64-darwin

### PR DESCRIPTION
**Pull request by azuwis**
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/soundfile/default.nix [hunk 1]
```diff
@@ -19,22 +19,28 @@ buildPythonPackage rec {
...
+  doCheck = stdenv.system != "aarch64-darwin";
+
+  meta = {
+    description = "An audio library based on libsndfile, CFFI and NumPy";
+    license = lib.licenses.bsd3;
+    homepage = "https://github.com/bastibe/python-soundfile";
+    maintainers = with lib.maintainers; [ fridh ];
+  };
+
+  postPatch = ''
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/soundfile/default.nix:39]: postPatch goes after patches.

File: pkgs/development/python-modules/soundfile/default.nix [hunk 2]
```diff
@@ -19,22 +19,28 @@ buildPythonPackage rec {
...
+    substituteInPlace soundfile.py --replace "_find_library('sndfile')" "'${libsndfile.out}/lib/libsndfile${stdenv.hostPlatform.extensions.sharedLibrary}'"
+  '';
+
+  # https://github.com/bastibe/PySoundFile/issues/157
+  disabled = isPyPy ||  stdenv.isi686;
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/soundfile/default.nix:44]: disabled should come after verison.

File: pkgs/development/python-modules/soundfile/default.nix [hunk 3]
```diff
@@ -19,22 +19,28 @@ buildPythonPackage rec {
...
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/soundfile/default.nix:45]:
```suggestion
```

File: pkgs/development/python-modules/soundfile/default.nix [hunk 4]
```diff
@@ -19,22 +19,28 @@ buildPythonPackage rec {
...
+
+  checkInputs = [ pytest ];
+  propagatedBuildInputs = [ numpy libsndfile cffi ];
+  propagatedNativeBuildInputs = [ cffi ];
+
+  # Test fails on aarch64-darwin with `MemoryError`, 53 failed, 31 errors, see
+  # https://github.com/bastibe/python-soundfile/issues/331
+  doCheck = stdenv.system != "aarch64-darwin";
+
+  meta = {
```
[the remark this answers is not in the document]
Comment (azuwis) [reply pkgs/development/python-modules/soundfile/default.nix:38]: Done

**Comment by azuwis** [comment]
All suggestions applied, please review the new changes.

File: pkgs/development/python-modules/soundfile/aarch64-darwin.patch [hunk 5]
```diff
@@ -0,0 +1,12 @@
+diff --git a/setup.py b/setup.py
+index b275081..517ef2a 100644
+--- a/setup.py
++++ b/setup.py
+@@ -16,6 +16,7 @@ MACOSX_VERSIONS = '.'.join([
+     'macosx_10_6_intel',
+     'macosx_10_9_intel',
+     'macosx_10_9_x86_64',
++    'macosx_11_0_arm64',
```
Comment (jonringer) [inline pkgs/development/python-modules/soundfile/aarch64-darwin.patch:9]: this isn't nix-specific, do you mind creating an upstream patch?

best would to then use `fetchpatch` to pull from the PR

Comment (azuwis) [reply pkgs/development/python-modules/soundfile/aarch64-darwin.patch:9]: Upstream are preparing a new 0.11 release https://github.com/bastibe/python-soundfile/issues/325, which reworked `setup.py` and fixed aarch64-darwin issue.

@bastibe Do you have any interest on this patch for 0.10 branch?

Comment (bastibe) [reply pkgs/development/python-modules/soundfile/aarch64-darwin.patch:9]: Dear @azuwis, I will make note of the MACOSX_VERSIONS you used here, as you seem to employ a test suite that can ensure their correctness. I have struggled to find a working set of platform identifiers on macOS, so this is a big help. Thank you!

Comment (azuwis) [reply pkgs/development/python-modules/soundfile/aarch64-darwin.patch]: > this isn't nix-specific, do you mind creating an upstream patch?
> 
> best would to then use `fetchpatch` to pull from the PR

Done, please review the new changes.

File: pkgs/development/python-modules/soundfile/default.nix [hunk 6]
```diff
@@ -12,29 +13,38 @@
 buildPythonPackage rec {
   pname = "soundfile";
   version = "0.10.3.post1";
+  # https://github.com/bastibe/PySoundFile/issues/157
+  disabled = isPyPy ||  stdenv.isi686;
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/soundfile/default.nix:17]:
```suggestion
  disabled = isPyPy || stdenv.isi686;
```

File: pkgs/development/python-modules/soundfile/default.nix [hunk 7]
```diff
@@ -12,29 +13,38 @@
 buildPythonPackage rec {
   pname = "soundfile";
   version = "0.10.3.post1";
+  # https://github.com/bastibe/PySoundFile/issues/157
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/soundfile/default.nix:16]:
```suggestion
  # https://github.com/bastibe/python-soundfile/issues/157
```

File: pkgs/development/python-modules/soundfile/default.nix [hunk 8]
```diff
@@ -12,29 +13,38 @@
...
-      maintainers = with lib.maintainers; [ fridh ];
-    };
+  postPatch = ''
+    substituteInPlace soundfile.py --replace "_find_library('sndfile')" "'${libsndfile.out}/lib/libsndfile${stdenv.hostPlatform.extensions.sharedLibrary}'"
+  '';
 
-    postPatch = ''
-      substituteInPlace soundfile.py --replace "_find_library('sndfile')" "'${libsndfile.out}/lib/libsndfile${stdenv.hostPlatform.extensions.sharedLibrary}'"
-    '';
+  checkInputs = [ pytest ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/soundfile/default.nix:36]:
```suggestion
  checkInputs = [ pytestCheckHook ];
```

**Comment by azuwis** [comment]
All suggestions applied, please review the new changes.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 161668` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>

  <summary>1 package failed to build and already failed to build on hydra master:</summary>
  <ul>
<li>python39Packages.elegy: log was empty</li>
  </ul>
</details>
<details>
<summary>
4 packages built:
</summary>
<ul>
<li>python310Packages.soundfile (python310Packages.pysoundfile)</li>
<li>python39Packages.coqui-trainer</li>
<li>python39Packages.librosa</li>
<li>python39Packages.soundfile (python39Packages.pysoundfile)</li>
</ul>
</details>

**Pull request merged**